src/libraries/Microsoft.Extensions.Logging.Console/src/WindowsLogConsole.cs

```diff
@@ -17,42 +17,108 @@ public WindowsLogConsole(bool stdErr = false)
         }
 
         private bool SetColor(ConsoleColor? background, ConsoleColor? foreground)
+        {
+            var backgroundChanged = SetBackgroundColor(background);
+            return SetForegroundColor(foreground) || backgroundChanged;
+        }
+
+        public bool SetBackgroundColor(ConsoleColor? background)
         {
             if (background.HasValue)
             {
                 System.Console.BackgroundColor = background.Value;
+                return true;
             }
+            return false;
+        }
 
+        public bool SetForegroundColor(ConsoleColor? foreground)
+        {
             if (foreground.HasValue)
             {
                 System.Console.ForegroundColor = foreground.Value;
+                return true;
             }
-
-            return background.HasValue || foreground.HasValue;
+            return false;
         }
 
         private void ResetColor()
         {
             System.Console.ResetColor();
         }
 
-        public void Write(string message, ConsoleColor? background, ConsoleColor? foreground)
+        public void Write(ReadOnlySpan<char> span, ConsoleColor? background, ConsoleColor? foreground)
         {
-            bool colorChanged = SetColor(background, foreground);
-            _textWriter.Write(message);
+            var colorChanged = SetColor(background, foreground);
+            _textWriter.Write(span);
             if (colorChanged)
             {
                 ResetColor();
             }
         }
 
-        public void WriteLine(string message, ConsoleColor? background, ConsoleColor? foreground)
+        public void Write(string message)
         {
-            bool colorChanged = SetColor(background, foreground);
-            _textWriter.WriteLine(message);
-            if (colorChanged)
+            // writes content with the embedded foreground and background ansi colored. it skips non-color related ansi codes
+            (int startIndex, int length, ConsoleColor? bg, ConsoleColor? fg) content = (-1, 0, null, null);
+            var span = message.AsSpan();
+            const char EscapeChar = '\x1B';
+            ConsoleColor? color = null;
+            bool isDarkColor = true;
+            for (int i = 0; i < span.Length; i++)
```

Comment:
Instead of doing the search one char at a time, we can use `message.IndexOf(EscapeChar)` which is much faster since it will use SIMD instructions to search multiple characters at a time. This should be faster